### DISCUSSION TITLE
How to read rest of the line into list of arguments ignoring options (even existing ones)

### DISCUSSION THREAD

**Discussion by petrsynek** [pallets/click#2531] [Q&A] [answered]
I want to read the rest of the line into the last argument ignoring that it contains even existing options (as the script that I am going to run may contain the same options as the main script containing click.

Using 
python 3.10.11
click 8.1.3

The best solution so far which unfortunately omits the `-d` option form `script_args` 

**launcher.py**
```python
import pathlib
import click

@click.command(
        context_settings=dict(
    ignore_unknown_options=True,
    )
)
@click.option(
    "-d",
    "--detach",
    is_flag=True,
)
@click.argument("script_path", nargs=1, type=click.Path(exists=True, path_type=pathlib.Path))
@click.argument("script_args", nargs=-1, type=click.UNPROCESSED)
def run(detach: bool, script_path: pathlib.Path, script_args: tuple[str]) -> None:

    print(f"Detach = {detach}")
    print(f"Script args = {script_args}")

if __name__ == "__main__":
    run()
```
when I run the script
```
> python launcher.py my_script.py -a -b -c -d -e
Detach = True
Script args = ('-a', '-b', '-c', '-e')
```
as you can see the '-d' flag is missing and yet '-d' flag was triggered which is undesired. I would expect ('-a', '-b', '-c', '-d', '-e') as a result.

What is the proper way within the click to do this?

**Comment by davidism** [comment, marked as answer]
Use `--` to separate options from tokens that must be treated as arguments.

```
python cli.py -a -b -c -- -d -e
```

**Reply by petrsynek** [reply]
Thanks a lot!

Follow-up question, is there any clear way how to communicate this need in --help (or should I explain in the function comment)?
